packages/frontend/app/components/NotificationBell.tsx

```diff
@@ -0,0 +1,404 @@
+
+  const uniqueCourseIds = useMemo(() => {
+    const ids = new Set<number>()
+    for (const alert of currentPageAlerts) {
+      if ((alert.payload as any)?.courseId) {
+        ids.add((alert.payload as any).courseId)
+      }
+    }
+    return Array.from(ids)
+  }, [currentPageAlerts])
+
+  const { data: courseResponses } = useSWR(
+    uniqueCourseIds.length > 0 ? ['courses-for-alerts', uniqueCourseIds] : null,
+    async () => {
+      const results: Record<number, string> = {}
+      await Promise.all(
+        uniqueCourseIds.map(async (id) => {
+          try {
+            const course: GetCourseResponse = await API.course.get(id)
+            results[id] = course?.name ?? `Course ${id}`
+          } catch {
+            results[id] = `Course ${id}`
+          }
+        }),
+      )
+      return results
+    },
+    { revalidateOnFocus: false },
+  )
+
+  const courseNameMap = courseResponses ?? {}
+
+  const markAsRead = async (alert: Alert) => {
+    if (alert.readAt) return
+    await markRead(alert.id)
+  }
+
+  const handleNavigate =
+    (alert: Alert, url: string) => async (): Promise<void> => {
+      await markAsRead(alert)
+      router.push(url)
+    }
+
+  const views: NotificationView[] = useMemo(() => {
+    return currentPageAlerts
+      .filter(
+        (alert) =>
+          alert.deliveryMode === AlertDeliveryMode.FEED &&
+          FEED_ALERT_TYPES.includes(alert.alertType),
+      )
+      .map((alert) => {
+        const sentAt = new Date(alert.sent)
+        const courseId = (alert.payload as any)?.courseId
+        const courseName = courseId ? courseNameMap[courseId] : undefined
+
+        if (alert.alertType === AlertType.DOCUMENT_PROCESSED) {
+          const payload = alert.payload as DocumentProcessedPayload
+          const destination = courseId
+            ? `/course/${courseId}/settings/chatbot_knowledge_base`
+            : undefined
+
+          return {
+            key: alert.id,
+            title: `Document "${payload.documentName}" is ready`,
+            description: 'Uploaded course document finished processing.',
+            ctaLabel: destination ? 'View document' : undefined,
+            onOpen: destination
+              ? handleNavigate(alert, destination)
+              : async () => await markAsRead(alert),
+            isUnread: !alert.readAt,
+            sent: sentAt,
+            courseName,
+          }
+        }
+
+        if (alert.alertType === AlertType.ASYNC_QUESTION_UPDATE) {
+          const payload = alert.payload as AsyncQuestionUpdatePayload
+          const destination = payload.courseId
+            ? `/course/${payload.courseId}/async_centre`
+            : undefined
+          const title =
+            payload.subtype === 'commentOnMyPost'
+              ? 'New comment on your Anytime Question'
+              : payload.subtype === 'commentOnOthersPost'
+                ? 'New comment on a followed Anytime Question'
```

Comment:
```suggestion
                ? 'New comment an Anytime Question you commented on'
```
Don't like using the term "follow" here since people might think they can "unfollow" it